tests/openkit-shared-tests/Core/Communication/BeaconSendingFlushSessionsStateTest.cs

```diff
@@ -82,20 +85,71 @@ public void ShutdownStateIsTerminalState()
         }
 
         [Test]
-        public void FinishedSessionsAreSent()
+        public void ToStringReturnStateName()
         {
-            // given 
-            finishedSessions.Enqueue(CreateValidSession("127.0.0.1"));
-            finishedSessions.Enqueue(CreateValidSession("127.0.0.1"));
+            // given
+            var target = new BeaconSendingFlushSessionsState();
+
+            // then
+            Assert.That(target.ToString(), Is.EqualTo("FlushSessions"));
+        }
 
-            httpClient.SendBeaconRequest(Arg.Any<string>(), Arg.Any<byte[]>()).Returns(x => new StatusResponse(string.Empty, 200));
+        [Test]
+        public void ABeaconSendingFlushSessionsStateTransitionsToTerminalStateWhenDataIsSent()
+        {
+            // given
+            var target = new BeaconSendingFlushSessionsState();
 
             // when
+            target.Execute(context);
+            
+            // then verify transition to terminal state
+            context.Received(1).NextState = Arg.Any<BeaconSendingTerminalState>();
+        }
+
+        [Test]
+        public void ABeaconSendingFlushSessionsStateConfiguresAllUnconfiguredSessions()
+        {
+            // given
             var target = new BeaconSendingFlushSessionsState();
+
+            var sessionOne = new SessionWrapper(CreateValidSession("127.0.0.1"));
+            var sessionTwo = new SessionWrapper(CreateValidSession("127.0.0.2"));
+            var sessionThree = new SessionWrapper(CreateValidSession("127.0.0.2"));
+            sessionThree.End();

```

Comment:
a comment that a session ended by the user is still unconfigured would ease the understanding why this statement is executed here